certbot-nginx/certbot_nginx/tests/configurator_test.py

```diff
@@ -401,6 +437,21 @@ def test_redirect_enhance(self):
         generated_conf = self.config.parser.parsed[example_conf]
         self.assertTrue(util.contains_at_depth(generated_conf, expected, 2))
 
+        # Test that we successfully add a redirect when there is
+        # no listen directive
+        migration_conf = self.config.parser.abs_path('sites-enabled/migration.com')
+        self.config.enhance("migration.com", "redirect")
+
+        generated_conf = self.config.parser.parsed[migration_conf]
+        self.assertTrue(util.contains_at_depth(generated_conf, expected, 2))
+
+    def test_redirect_dont_enhance(self):
+        with LogCapture() as l:
```

Comment:
While this is definitely nicer for the developer than what I'm about to propose, I think we should remove `testfixtures` from the project. Many of our package maintainers use our unit tests to check if Certbot is working properly on their system. When they run these tests, they do so without our `dev` dependencies meaning this test would fail for them. We could add `testfixtures` as a regular dependency but that is a step backwards on #1301 and requires all OS's to have a `testfixtures` package available.

Instead, I think we should just use `mock` here. You can probably register logging handler and check output that way, but for this test you can do something like:
```
        with mock.patch("certbot_nginx.configurator.logger") as mock_logger:
            self.config.enhance("geese.com", "redirect")
        self.assertEqual(mock_logger.info.call_args[0][0],
                'No matching insecure server blocks listening on port %s found.')
```
Note the need to use `%s` rather than the port number.